R/install.R

```diff
@@ -344,19 +339,19 @@ install <-
     cmp <- .version_compare(version, version())
     action <- if (cmp < 0) "Downgrade" else "Upgrade"
     repos <- repositories(site_repository, version = version)
-    biocrepos <- repos[names(repos) != "CRAN"]
 
     if (cmp != 0L) {
         pkgs <- unique(c("BiocVersion", pkgs))
-        npkgs <- .resolve_npkgs(inst, biocrepos)
+        valist <- suppressWarnings(.valid(version = version))
```

Comment:
Better to move the warning out of `.valid()` and into `valid()`, so that unrelated warnings are not suppressed; this is a general best practice -- avoid triggering known warning / error conditions rather than trying to catch & recover.